### DISCUSSION TITLE
Trying to create a local development environment for nextjs-chat project and receiving the following error when installing dependencies

### DISCUSSION THREAD

**Discussion by tompongrac** [vercel/next.js#59803] [Help]
### Summary

Hi everybody. I am very new to vercel and am starting my first project. I wish to develop it on my localhost so I have cloned the repository, installed npm, node 18.19.0 'Hydrogen' and now am trying to install the nextjs-chat projects dependencies using 'npm install'. Please note my operating system is Windows 10 and Im running it from PS from within Visual Studio Code.

`PS C:\Users\alexm\Documents\Websites\nextjs-chat> npm install
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: undefined@undefined
npm ERR! Found: next@14.0.4-canary.17
npm ERR! node_modules/next
npm ERR!   next@"14.0.4-canary.17" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer next@"^13.2 || ^14" from geist@1.2.0
npm ERR! node_modules/geist
npm ERR!   geist@"^1.1.0" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
npm ERR!
npm ERR! For a full report see:
npm ERR! C:\Users\alexm\AppData\Local\npm-cache\_logs\2023-12-20T11_12_14_322Z-eresolve-report.txt

npm ERR! A complete log of this run can be found in: C:\Users\alexm\AppData\Local\npm-cache\_logs\2023-12-20T11_12_14_322Z-debug-0.log`

The error log they are referring to contains
`'Log files:
C:\Users\alexm\AppData\Local\npm-cache\_logs\2023-12-20T11_12_14_322Z-debug-0.log

# npm resolution error report

While resolving: undefined@undefined
Found: next@14.0.4-canary.17
node_modules/next
  next@"14.0.4-canary.17" from the root project

Could not resolve dependency:
peer next@"^13.2 || ^14" from geist@1.2.0
node_modules/geist
  geist@"^1.1.0" from the root project

Fix the upstream dependency conflict, or retry
this command with --force or --legacy-peer-deps
to accept an incorrect (and potentially broken) dependency resolution.
`

They have instructed to me that to resolve the conflict I could use the --force or --legacy-peer-deps command line argument, however I do not know if this is a safe and stable way to solve this 'dependency resolution'. If anyone knows if I can safely force it or possibly knows the correct way to fix this error it would be highly appreciated.

Thanks in advance!
Tom P

**Comment by ananuness** [comment]
Hii Tom!! When this happens, it's because of unmatched dependencies with your environment like you're using the latest next but the project has deps using an older one... As the flags they suggested, it's ok to run it but they warn you that there could be incorrect and broken deps! 

So, you can run with `--force` or `--legacy-peer-deps` and see if it works! Although I would recommend to use those projects as inspiration 😊
